### PR TITLE
Updating CodeAnalysis Nuget

### LINKED ISSUE #2053: Build Error: StandardPerformanceCounter IDisposable
Build system is throwing this error intermittently. Need to confirm this change is made in the code.

> ##[error]WEB\Src\PerformanceCollector\PerformanceCollector\Implementation\StandardPerformanceCollector\StandardPerformanceCounter.cs(12,36): 
> Error CA2213: 'StandardPerformanceCounter' contains field 'performanceCounter' that is of IDisposable type 'PerformanceCounter', but it is never disposed. 
> Change the Dispose method on 'StandardPerformanceCounter' to call Close or Dispose on this field.

### PR DESCRIPTION
**Pull request by eddynaka**
Fix #2053

## Changes
(Please provide a brief description of the changes here.)

### Checklist
- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Review by TimothyMothra — APPROVED** [review]

File: .rulesets/ApplicationInsightsSDKRules.ruleset [hunk 1]
```diff
@@ -31,7 +31,13 @@
     <Rule Id="CA1410" Action="Warning" />
     <!-- Suppressing Message "Identifiers should not match keywords" because class TplActivities is part of our public api and can't be changed -->
     <Rule Id="CA1716" Action="None" />
+    <!-- Errors shown after FxCop upgrade to 3.3.0. Turning off CA1805 to ignore these errors-->
```
Comment (TimothyMothra) [inline .rulesets/ApplicationInsightsSDKRules.ruleset:34]: This comment doesn't tell me what the rule is or why it was disabled.
See the comment on line 32

**Comment by TimothyMothra** [comment]
Add those two TODOs to the original Issue. We can work those in a follow up PR, and get this merged today

**Comment by eddynaka** [comment]
> Add those two TODOs to the original Issue. We can work those in a follow up PR, and get this merged today

I solved the major ones and added the TODOs where I couldn't do it. We are good to go

**Comment by TimothyMothra** [comment]
I mean add the file name and line number to the original issue so we can circle back and address them

**Comment by eddynaka** [comment]
> I mean add the file name and line number to the original issue so we can circle back and address them

Just updated the main issue. thank you.

**Review by TimothyMothra — APPROVED** [review]

**Pull request merged**